zadanie zrobione

jtermi opened the pull request.

jtermi committed:
zadanie zrobione

marcinkunert commented on src/main/java/pl/javastart/task/Person.java:
```diff
@@ -0,0 +1,75 @@
+
+    public Person(String firstName, String lastName, int age, String pesel) {
+
+        if (firstName == null || lastName == null || firstName.length() < 2 || lastName.length() < 2) {
+            throw new NameUndefinedException();
+        }
+
+        if (age < 1) {
+            throw new IncorrectAgeException();
+        }
```
Te same warunki masz w setterach. Postaraj się przerobić kod tak, żeby ich nie powtarzać.

jtermi committed:
poprawilem bledy
- utworzylem metody validacyjne z ktorych korzystam zarowno w konstruktorze jak i seterach.

slawekludwiczak commented on src/main/java/pl/javastart/task/Person.java:
```diff
@@ -0,0 +1,81 @@
+    private String pesel;
+
+    public Person(String firstName, String lastName, int age, String pesel) {
+        validateFirstName(firstName);
+        validateLastName(lastName);
+        validateAge(age);
+        this.firstName = firstName;
+        this.lastName = lastName;
+        this.age = age;
```
Tutaj możesz wykorzystać settery, dzięki czemu kod będzie krótszy i unikniesz powtarzania logiki z setterów
```
public Person(String firstName, String lastName, int age, String pesel) {
    setFirstName(firstName);
    setLastName(lastName);
    setAge(age);
    this.pesel = pesel;
}
```

slawekludwiczak reviewed (changes requested):
code review https://javastartbootcamp.slack.com/archives/C04JX1NL08G/p1680251386949839
